### PR TITLE
refactor(orca-api): move ExecutionPreprocessor and TaskExecutionInterceptor into orca-api

### PR DESCRIPTION
**Pull request by aelath**
Moving two existing interfaces from `orca-core` into `orca-api`. The purpose is to communicate that these interfaces are safe to use for extending orca functionality.
- `ExecutionPreprocessor`
- `TaskExecutionInterceptor`

### CODE REVIEW

File: orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java [hunk 1]
```diff
@@ -13,12 +13,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.netflix.spinnaker.orca.extensionpoint.pipeline;
+package com.netflix.spinnaker.orca.api.pipeline;
 
-import java.util.Map;
 import javax.annotation.Nonnull;
+import java.util.Map;
```
Comment (danielpeach) [inline orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java:19]: Should this sort above `javax.annotation.Nonnull`?

File: orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java [hunk 2]
```diff
@@ -13,12 +13,18 @@
...
 
-/** A preprocessor that can modify an Execution upon initial receipt of the configuration. */
+/**
+ * ExecutionPreprocessor is a hook point that can modify an Execution upon initial receipt
+ * of the configuration.
+ *
+ * <p>A common use case is to inspect and update a pipeline with configured context, such
+ * an application name.
+ */
 public interface ExecutionPreprocessor {
```
Comment (danielpeach) [inline orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java:28]: Do you want to make this a `SpinnakerExtensionPoint`?

Comment (danielpeach) [reply orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java:28]: I would mark the interface as `@Beta`

**Review by danielpeach — APPROVED** [review]

File: orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/TaskExecutionInterceptor.java [hunk 3]
```diff
@@ -14,10 +14,8 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.orca;
+package com.netflix.spinnaker.orca.api.pipeline;
 
-import com.netflix.spinnaker.orca.api.pipeline.Task;
-import com.netflix.spinnaker.orca.api.pipeline.TaskResult;
 import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
 import java.util.concurrent.TimeUnit;
```
Comment (danielpeach) [inline orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/TaskExecutionInterceptor.java:20]: Sorry - more stuff:

- Could you mark this interface as `@Beta`?
- Any reason not to have it extend `SpinnakerExtensionPoint`?
- Could you add doc comments to the methods that aren't documented?

Comment (aelath) [reply orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/TaskExecutionInterceptor.java:20]: I don't think there is any reason to not have it extend `SpinnakerExtensionPoint`, thats a good idea. I'll also mark it as `@Beta` and add doc comments. 👍

**Comment by aelath** [comment]
TODO:

`ExecutionPreprocessor.java`

- [ ] sort imports
- [ ] add `@Beta` annotation
- [ ] extend `SpinnakerExtensionPoint`

`TaskExecutionInterceptor.java`

- [ ] add doc comments for undocumented methods
- [ ] add `@Beta` annotation
- [ ] extend `SpinnakerExtensionPoint`

**Review by danielpeach — APPROVED** [review]

**Pull request merged**